controllers/nnf_workflow_controller_helpers.go

```diff
@@ -1089,7 +1097,7 @@ func (r *NnfWorkflowReconciler) createOrUpdateContainerJobsIfNecessary(ctx conte
 				podSpec.RestartPolicy = corev1.RestartPolicyNever
 
 				// Because the IP changes, we need to have deterministic hostname for the pod
-				podSpec.Hostname = jobName
+				podSpec.Hostname = nnfNode
```

Comment:
This should be ok since the service name (subdomain) is the workflow name, right?

Reply:
The pod and service are created in the workflow namespace, so that should be `nnf-node-name.service-name.workflow-namespace.svc.cluster.local`